src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1099,16 +1099,54 @@ MavlinkReceiver::handle_message_set_actuator_control_target(mavlink_message_t *m
 
 			actuator_controls.timestamp = hrt_absolute_time();
 
-			/* Set duty cycles for the servos in actuator_controls_0 */
+			/* Set duty cycles for the servos in the actuator_controls message */
 			for (size_t i = 0; i < 8; i++) {
 				actuator_controls.control[i] = set_actuator_control_target.controls[i];
 			}
 
-			if (_actuator_controls_pub == nullptr) {
-				_actuator_controls_pub = orb_advertise(ORB_ID(actuator_controls_0), &actuator_controls);
+			switch (set_actuator_control_target.group_mlx) {
+			case 0:
+				if (_actuator_controls_pubs[0] == nullptr) {
```

Comment:
Once last nitpicky thing, can you change these into orb_publish_auto? It includes the nullptr check and initiate advertise in a single call.
https://github.com/PX4/Firmware/blob/master/src/modules/uORB/uORB.h#L177

Reply:
``` C
    int inst
    orb_publish_auto(ORB_ID(actuator_controls_0), &_actuator_controls_pubs[0], &actuator_controls, inst, ORB_PRIO_DEFAULT)
```